### DISCUSSION TITLE
model import on html5 runtime?

### DISCUSSION THREAD

**Discussion by ElhamAryanpur** [godotengine/godot-proposals#4038] [3D]
I have been working on a project that requires importing and placing models from external sources. However I have been reading and got working through instructions and work Mr. iFire did, but what I have not been successful was having it on html5 which is the only export the project requires.

All source models for the project is gltf 2.0, and I am not sure how to fix/make it work on html5. I am running Godot 3.4.2.

**Comment by Calinou** [comment]
See https://github.com/godotengine/godot-proposals/issues/3273. This feature will only be available in Godot 4.0, *not* Godot 3.5 to avoid breaking compatibility with existing projects.

However, Godot 4.0 will most likely only support basic 2D rendering in HTML5, due to WebGPU not being mature for production use yet (and no WebGPU backend being written for Godot yet). Therefore, only the 2D-only OpenGL renderer will be usable in HTML5 (with a [3D OpenGL renderer](https://github.com/godotengine/godot-proposals/issues/3959) planned for 4.x).

**Reply by ElhamAryanpur** [reply]
oh damn... any workarounds or alternative ways? I have tried module by V-Sekai but it doesn't support html5 either...

**Reply by Calinou** [reply]
You could use a [run-time OBJ loader](https://github.com/Ezcha/gd-obj) if you only need to import 3D models that don't have any animations. However, writing a glTF importer in GDScript would be a lot of work – it's not a simple specification.

**Reply by ElhamAryanpur** [reply]
thanks, I'll try obj loader.
